fog/ledger/server/tests/store_tests.rs

```diff
@@ -0,0 +1,326 @@
+    let auth_response_event = AuthResponseInput::new(auth_message.into(), Verifier::default());
+    // Should be a valid noise connection at this point.
+    let (mut noise_connection, _verification_report) =
+        initiator.try_next(&mut rng, auth_response_event).unwrap();
+
+    //Construct our request.
+    let key_images_request = CheckKeyImagesRequest {
+        queries: vec![KeyImageQuery {
+            key_image: test_key_image.key_image,
+            start_block: 1,
+        }],
+    };
+    // Protobuf-encoded plaintext.
+    let message_encoded = mc_util_serial::encode(&key_images_request);
+    let ciphertext = noise_connection.encrypt(&[], &message_encoded).unwrap();
+    let msg: EnclaveMessage<ClientSession> = EnclaveMessage {
+        aad: vec![],
+        channel_id: client_session,
+        data: ciphertext,
+    };
+
+    // Decrypt and seal
+    let sealed_query = enclave.decrypt_and_seal_query(msg).unwrap();
+    println!(
+        "Client session on sealed_query is {:?}",
+        &sealed_query.channel_id
+    );
+    let mut multi_query = enclave
+        .create_multi_key_image_store_query_data(sealed_query.clone())
+        .unwrap();
+
+    let query: EnclaveMessage<NonceSession> = multi_query.pop().unwrap();
+    println!("Nonce session on message is {:?}", query.channel_id);
+
+    // Get an untrusted query
+    let (
+        highest_processed_block_count,
+        last_known_block_cumulative_txo_count,
+        latest_block_version,
+    ) = {
+        let shared_state = shared_state.lock().expect("mutex poisoned");
+        (
+            shared_state.highest_processed_block_count,
+            shared_state.last_known_block_cumulative_txo_count,
+            shared_state.latest_block_version,
+        )
+    };
+
+    let untrusted_kiqr = UntrustedKeyImageQueryResponse {
+        highest_processed_block_count,
+        last_known_block_cumulative_txo_count,
+        latest_block_version,
+        max_block_version: latest_block_version.max(*MAX_BLOCK_VERSION),
+    };
+
+    let result = enclave
+        .check_key_image_store(query, untrusted_kiqr)
+        .unwrap();
+
+    let responses_btree: BTreeMap<ResponderId, EnclaveMessage<NonceSession>> =
+        BTreeMap::from([(responder_id, result)]);
+
+    let client_response = enclave
+        .collate_shard_query_responses(sealed_query, responses_btree)
+        .unwrap();
+
+    let plaintext_bytes = noise_connection
+        .decrypt(&client_response.aad, &client_response.data)
+        .unwrap();
+
+    let done_response: CheckKeyImagesResponse = mc_util_serial::decode(&plaintext_bytes).unwrap();
+    assert_eq!(done_response.results.len(), 1);
+
+    let mut test_results = done_response
+        .results
+        .into_iter()
+        .map(|result| (result.key_image, result.key_image_result_code));
+
+    // The key image result code for a spent key image is 1.
+    assert!(test_results.contains(&(test_key_image.key_image, 1)));
```

Comment:
:wrench: Suggest comparing to a slice. It provides nicer output on failures and ensures both the length and contents
```suggestion
    let test_results = done_response
        .results
        .into_iter()
        .map(|result| (result.key_image, result.key_image_result_code))
        .collect::<Vec<_>>();

    // The key image result code for a spent key image is 1.
    assert_eq!(test_results, &[(test_key_image.key_image, 1)]);
```